src/stdlib_experimental_error.f90

```diff
@@ -1,41 +1,34 @@
 module stdlib_experimental_error
+use, intrinsic :: iso_fortran_env, only: stderr=>error_unit
```

Comment:
This does not need to be imported here, does it?

Reply:
I tested this patch and it works:
```diff
diff --git a/src/f08estop.f90 b/src/f08estop.f90
index d501978..17ea448 100644
--- a/src/f08estop.f90
+++ b/src/f08estop.f90
@@ -1,4 +1,5 @@
 submodule (stdlib_experimental_error) estop
+use, intrinsic :: iso_fortran_env, only: stderr=>error_unit
 
 contains
 
diff --git a/src/f18estop.f90 b/src/f18estop.f90
index ea83de7..6be7900 100644
--- a/src/f18estop.f90
+++ b/src/f18estop.f90
@@ -1,4 +1,5 @@
 submodule (stdlib_experimental_error) estop
+use, intrinsic :: iso_fortran_env, only: stderr=>error_unit
 
 contains
 
diff --git a/src/stdlib_experimental_error.f90 b/src/stdlib_experimental_error.f90
index 3d932d6..135722b 100644
--- a/src/stdlib_experimental_error.f90
+++ b/src/stdlib_experimental_error.f90
@@ -1,5 +1,4 @@
 module stdlib_experimental_error
-use, intrinsic :: iso_fortran_env, only: stderr=>error_unit
 implicit none
 private
 ```

Reply:
it's more of a deduplication issue. The scoping of submodule is such that these submodules pickup `stderr` from the parent module `stdlib_experimental_error` e.g. imagine there are dozens of submodules (which eventually there will be in this project) they don't all need to each import the same modules, just import once in the parent or ancestor module.

Reply:
Ok, that's fine. I don't use submodules myself, so I don't know what the best way to use them is. I thought the idea is that the main module is sort of like a C++ `.h` file and the submodule is then like `.cpp` file. In C++, the accepted practice is to only include things in the `.h` files that are needed to specify the API. Everything else is included in the `.cpp` files. In the same spirit, since `stderr` is not needed for the API, I thought it would make sense to only include it in the submodule.

Reply:
One could use module/submodule that way, but they are generally not used that way as the scoping rules are different than C++. I use submodules for almost every program for reasons including:

* very fast rebuild if procedure API didn't change, can be 100x or more build wallclock speed, important for big programs that take several minutes to build fully, but only a couple seconds if rebuilding after submodule change
* allows switching in/out capabilities or libraries. Examples
  * file IO with HDF5, NetCDF4 or fallback to raw binary if they're not available, switched by the build systems auto/manual
  * external procedures/modules, maybe some are proprietary or take a long build time or a lot of prereqs, make them optional yet almost instantly switchable in/out via build system
  * procedure switching, maybe the procedures are overloaded in certain ways as defined by a simulation, again making almost instant rebuilds on big programs with build system options

So submodule can be used like one step up from "just in time" compilation, I call it "build on run" since based on options say called from Python API, I can rebuild huge Fortran program based on options user specifies from Python (or directly in CMake) in seconds.

Reply:
submodule can also be used for build-time polymorphism, avoiding need for preprocessing and again shrinking memory use and build time by only building the required `kind`s.
I believe a significant part of the preprocessing and generation seen in Fortran could be replaced more cleanly and readably with `submodule`, `select type` and `select rank`. Of course will still be good applications for a small `#ifdef foo #endif` and generation.